include/onnxruntime/core/session/environment.h

```diff
@@ -86,18 +100,76 @@ class Environment {
    * Return an error if an allocator with the same OrtMemoryInfo is already registered.
    * For provider_type please refer core/graph/constants.h
    */
-  Status CreateAndRegisterAllocatorV2(const std::string& provider_type, const OrtMemoryInfo& mem_info, const std::unordered_map<std::string, std::string>& options, const OrtArenaCfg* arena_cfg = nullptr);
+  Status CreateAndRegisterAllocatorV2(const std::string& provider_type, const OrtMemoryInfo& mem_info,
+                                      const std::unordered_map<std::string, std::string>& options,
+                                      const OrtArenaCfg* arena_cfg = nullptr);
+
+  Status RegisterExecutionProviderLibrary(const std::string& registration_name, const ORTCHAR_T* lib_path);
+  Status UnregisterExecutionProviderLibrary(const std::string& registration_name);
```

Comment:
Can these use std::string_view?

Reply:
they could, but it doesn't end up buying you anything as it needs to be std::string to do the internal lookups like `ep_libraries_count(registration_name)`. we need a std::string in the key of that to guarantee it remains valid. 

so if I start with a std::string there's an indirection to string_view and I have to construct a new std::string inside RegisterExecutionProviderLibrary (for example), costing me at least one extra std::string allocation.